pkg/node/manager/manager_test.go

```diff
@@ -833,3 +840,51 @@ func (s *managerTestSuite) TestNode(c *check.C) {
 	// Needs to be the same as n2
 	c.Assert(n, checker.DeepEquals, *n1V2)
 }
+
+func TestNodeManagerEmitStatus(t *testing.T) {
+	// Tests health reporting on node manager.
+	assert := assert.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	baseBackgroundSyncInterval = time.Millisecond * 10
+	hr := &hivetest.MockHealthReporter{}
+	m, err := New(&configMock{}, newIPcacheMock(), NewNodeMetrics(), hr)
+	assert.NoError(err)
+
+	m.nodes[nodeTypes.Identity{
+		Name:    "node1",
+		Cluster: "c1",
+	}] = &nodeEntry{node: nodeTypes.Node{Name: "node1", Cluster: "c1"}}
+	m.nodeHandlers = make(map[datapath.NodeHandler]struct{})
+	nh1 := newSignalNodeHandler()
+	nh1.EnableNodeValidateImplementationEvent = true
+	nh1.NodeValidateImplementationEventError = fmt.Errorf("test error")
+	m.nodeHandlers[nh1] = struct{}{}
+
+	// Start the manager
+	assert.NoError(m.Start(context.Background()))
+	time.Sleep(time.Millisecond * 100)
```

Comment:
Why we need this `time.Sleep` here? I am afraid that this might increase test flakyness when running in CI.